### PR TITLE
new_post/react-localstorage

### PR DESCRIPTION
**Pull request by cr0wg4n**
To review :D

### CODE REVIEW

File: source/_posts/react-localstorage.md [hunk 1]
```diff
@@ -0,0 +1,382 @@
+---
+title: Using localStorage with React Hooks
+contributor: Mauricio Matias C.
+avatar: mauricio-matias.jpeg
+description: Learn how to use localStorage and React with examples and make your own implementation decisions. 
+date: 2022-12-22
+tags: [react]
+id: react-localstorage
+relatedPost: [react-security-best-practices, react-xss]
```
Comment (bartosz-io) [inline source/_posts/react-localstorage.md:9]: Should be only 1 related post.

File: source/_posts/react-localstorage.md [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+---
+
```
Comment (bartosz-io) [inline source/_posts/react-localstorage.md:11]: Please remove that empty line.

File: source/_posts/react-localstorage.md [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+{% image_fw 1.78 banner.png "React localStorage" %}
+
+Hi, React developers! Today, we will learn how to store data using `localStorage`, and some uses of it to mix up in your React App. So, when do we need to use local storage? The short and simple answer is for data persistence, but it doesn't mean that all your app's data should persist. We need some criteria to store portions of information.
+
+## Table of Contents
+<!-- toc -->
+
+## localStorage
+
+The `localStorage` is the read-only property of the window interface for storing data across browser sessions (in other words, browser storage). The `localStorage` API is compatible with many modern browsers, is one of two ways to store data locally (client side), and the maximum volume of information that `localStorage` stores are 5 MB. The other one is `sessionStorage`. It only stores data while the session is active. Otherwise, `localStorage` doesn't have an expiration date even if the browser is closed or the OS reboots. That is the magic power of `localStorage`, and that's what we talk about today.
```
Comment (bartosz-io) [inline source/_posts/react-localstorage.md:21]: window interface? `window` is an object I think ;)

Comment (bartosz-io) [reply source/_posts/react-localstorage.md:21]: Would make sense to add the info about **Web Storage API** standard.

File: source/_posts/react-localstorage.md [hunk 4]
```diff
@@ -0,0 +1,382 @@
...
+// Alternativelly 'delete' works but it happens thanks 
+// to the language and not of the localStorage implementation
+// so be careful, we don't recommend its use
+delete localStorage.counter;
+
+// Cleaning all data from localStorage 
+localStorage.clear();
+```
+
+`removeItem` and clear are different. `removeItem` removes a key-value pair, and `clear` removes all the data inside `localStorage`.
```
Comment (bartosz-io) [inline source/_posts/react-localstorage.md:109]: `clear` ?

Comment (bartosz-io) [reply source/_posts/react-localstorage.md:109]: or even better `removeItem()` and `clear()` ?

File: source/_posts/react-localstorage.md [hunk 5]
```diff
@@ -0,0 +1,382 @@
...
+        Button 2: count is {secondCounter}
+      </button>
+    </div>
+  )
+}
+```
+
+{% img "hook_example.png" "Hook react component, react localstorage, hook style" "lazy" %}
+
+See how the code reduces a lot and how we can reuse the hook (many times as we need).
```
Comment (bartosz-io) [inline source/_posts/react-localstorage.md:254]: "See how the code reduces a lot" hmm does not sound good grammar-wise :D code shrinks (itself) maybe rather reduces (something - what?) ?

File: source/_posts/react-localstorage.md [hunk 6]
```diff
@@ -0,0 +1,382 @@
...
+
+### Miscellaneous Usages
+
+To finish this implementation section. We have a miscellaneous use of `localStorage`. Usually, the data from `localStorage` is accessible, and every user can see it as plain text (because it is string based), but if we need to hide it from users, there is a way to encrypt it. We need to use the [encrypt-storage](https://github.com/michelonsouza/encrypt-storage) library.
```
Comment (bartosz-io) [inline source/_posts/react-localstorage.md:258]: **string-based** not **string based**

File: source/_posts/react-localstorage.md [hunk 7]
```diff
@@ -0,0 +1,382 @@
...
+
+Before we need to configure that encrypted storage, for that create a new js file like this:
```
Comment (bartosz-io) [inline source/_posts/react-localstorage.md:260]: **Before,** with **,** ?

Comment (bartosz-io) [reply source/_posts/react-localstorage.md:260]: new `.js` file ?

File: source/_posts/react-localstorage.md [hunk 8]
```diff
@@ -0,0 +1,382 @@
...
+      <br /><br />
+      <button onClick={() => setSecondCounter(secondCounter + 1)}>
+        Button 2: encrypted count is {secondCounter}
+      </button>
+    </div>
+  )
+}
+```
+
+Use the `developer tools` go to Application>Local Storage tool and check the `localStorage`. The stored data is currently encrypted 😎 and is prefixed with our selected prefix (`enc` in this case).
```
Comment (bartosz-io) [inline source/_posts/react-localstorage.md:359]: `developer tools` - is that some syntax to be highlighted? :D

**Review by bartosz-io — COMMENTED** [review]
I don't mind changing your profile picture, but in my opinion profile pictures where you **look straight in front of you** seem more trustworthy compared to side-face or when the face is somehow hidden :)

File: source/_posts/react-localstorage.md (hunk 5 above)
Comment (bartosz-io) [reply source/_posts/react-localstorage.md:254]: Or _the amound code IS reduced_ ?

File: source/_posts/react-localstorage.md (hunk 1 above)
Comment (cr0wg4n) [reply source/_posts/react-localstorage.md]: Done :D

File: source/_posts/react-localstorage.md (hunk 2 above)
Comment (cr0wg4n) [reply source/_posts/react-localstorage.md]: Done :D

File: source/_posts/react-localstorage.md (hunk 4 above)
Comment (cr0wg4n) [reply source/_posts/react-localstorage.md]: done :D

File: source/_posts/react-localstorage.md (hunk 5 above)
Comment (cr0wg4n) [reply source/_posts/react-localstorage.md]: Done :D

File: source/_posts/react-localstorage.md (hunk 6 above)
Comment (cr0wg4n) [reply source/_posts/react-localstorage.md]: Done :D

File: source/_posts/react-localstorage.md (hunk 7 above)
Comment (cr0wg4n) [reply source/_posts/react-localstorage.md]: Done :D

File: source/_posts/react-localstorage.md (hunk 8 above)
Comment (cr0wg4n) [reply source/_posts/react-localstorage.md]: Done :D

**Review by bartosz-io — APPROVED** [review]

**Pull request merged**